### PR TITLE
Fix depagination in ghutil package

### PR DESCRIPTION
**Pull request by chaodaiG**
Currently depagination keeps on querying the first page, due to ListOptions not passed in as reference while querying. Passing in correct ListOptions to make it work

### CODE REVIEW

File: shared/ghutil/issue.go [hunk 1]
```diff
@@ -63,20 +63,19 @@ func (gc *GithubClient) ListRepos(org string) ([]string, error) {
...
 		issueListOptions.Labels = labels
 	}
 
-	options := &github.ListOptions{}
 	genericList, err := gc.depaginate(
 		fmt.Sprintf("listing issues with label '%v'", labels),
 		maxRetryCount,
-		options,
+		&issueListOptions.ListOptions,
 		func() ([]interface{}, *github.Response, error) {
```
Comment (adrcunha) [inline shared/ghutil/issue.go]: I can't help thinking that you could simplify depaginate even more, by encapsulating this repeated pattern (call gc.Client/for/append/return) into it and just passing a function that calls gc.Client.X. But I won't block on it.

Comment (chaodaiG) [reply shared/ghutil/issue.go]: I think you're absolutely right. I will let this PR in first, and see what I can do refactoring this in another PR

Comment (chaodaiG) [reply shared/ghutil/issue.go]: I have done some investigation, the sad news is this is almost not possible, due to the lack of generic support in Go. As in this case, `ListByRepo` function can not be passed in as a  parameter as the arguments vary among other `List` functions. And this line `page, resp, err := gc.Client.Issues.ListByRepo(ctx, org, repo, issueListOptions)` can not be used as an argument, as the return type(`page`) is different among other `List` functions. So seems like converting `page` to an interface slice locally is not avoidable. @adrcunha

Comment (adrcunha) [reply shared/ghutil/issue.go]: :(

**Pull request merged**